docs/proposals/status-aggregation.md

```diff
@@ -10,39 +10,127 @@ When multiple instances of a resource (e.g. a Gateway) are running across multip
 
-The `lastTransitionTime` and `observedGeneration` fields will behave as normal for the resource in the control plane.
\ No newline at end of file
+The `lastTransitionTime` and `observedGeneration` fields will behave as normal for the resource in the control plane.
+
+### Addresses and Listeners status
+
+The Gateway status can include information about addresses, like load balancer IP Addresses assigned to the Gateway,
+and listeners, such as the number of attached routes for each listener.
+This information is useful at the control plane level.
+For example, a DNS Record should only exist as long as there is at least 1 attached route for a listener.
+It can also be more complicated than that when it comes to multi cluster gateways.
+A DNS Record should only include the IP Addresses of the Gateway instances where the listener has at least 1 attached route.
+This is important when initial setup of DNS Records happen as applications start.
+It doesn't make sense to route traffic to a Gateway where a listener isn't ready/attached yet.
+It also comes into play when a Gateway is displaced either due to changing placement decision or removal.
+
+In summary, the IP Addresses and number of attached routes per listener per Gateway instance is needed in the control plane to manage DNS effectively.
+This proposal adds that information the hub Gateway status block.
+This will ensure a decoupling of the DNS logic from the underlying resource/status syncing implementation (i.e. ManifestWork status feedback rules)
+
+First, here are 2 instances of a multi cluster Gateway in 2 separate spoke clusters.
+The yaml is shortened to highlight the status block.
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: gateway
+status:
+  addresses:
+  - type: IPAddress
+    value: 172.32.200.0
+  - type: IPAddress
+    value: 172.32.201.0
+  listeners:
+  - attachedRoutes: 0
+    conditions:
+    name: api
+  - attachedRoutes: 1
+    conditions:
+    name: web
+---
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: gateway
+status:
+  addresses:
+  - type: IPAddress
+    value: 172.32.202.0
+  - type: IPAddress
+    value: 172.32.203.0
+  listeners:
+  - attachedRoutes: 1
+    name: api
+  - attachedRoutes: 1
+    name: web
+```
+
+And here is the proposed status aggregation in the hub Gateway:
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: gateway
+status:
+  gateways:
+  - addresses:
+    - type: IPAddress
+      value: 172.32.200.0
+    - type: IPAddress
+      value: 172.32.201.0
+    listeners:
+    - attachedRoutes: 0
+      name: api
+    - attachedRoutes: 1
+      name: web
+  - addresses:
+    - type: IPAddress
+      value: 172.32.202.0
+    - type: IPAddress
+      value: 172.32.203.0
+    listeners:
+    - attachedRoutes: 1
+      name: api
+    - attachedRoutes: 1
+      name: web
+```
```

Comment:
In this example it has addresses from 2 different clusters, but there is no way to map back the ips to a cluster right?

This is probably good enough to ensure that we are updating this gateway when DNS needs to be updated, but it doesn't contain enough information for the DNS reconciler to fully work without querying the ocm placement api to get the addresses per cluster. Wondering if there is anything we can do here to surface the cluster name for each group of addresses?

Reply:
> In this example it has addresses from 2 different clusters, but there is no way to map back the ips to a cluster right?

Yes. There's no cluster or gateway instance identifier.


> Wondering if there is anything we can do here to surface the cluster name for each group of addresses

What kind of structure is needed/used currently?
Is it a map of cluster names to ip addresses?

Reply:
We use the placement decision to find which clusters and then a manifestwork to find the cluster ips